pandas/tests/frame/test_asof.py

```diff
@@ -106,3 +106,30 @@ def test_all_nans(self):
         result = DataFrame(np.nan, index=[1, 2], columns=['A', 'B']).asof(3)
         expected = Series(np.nan, index=['A', 'B'], name=3)
         tm.assert_series_equal(result, expected)
+
+    def test_time_zone_aware_index(self):
```

Comment:
If possible, could you use `pytest.mark.parameterize` to combine these two tests? The structure of these tests seem very similar. Looks like you might be able to param over `timestamp_internal`